workflow/controller/operator.go

```diff
@@ -1160,6 +1179,12 @@ func (woc *wfOperationCtx) podReconciliation(ctx context.Context) error {
 
 	wg.Wait()
 
+	// If true, it means there are some nodes which have outputs want to be mark succeed, but the node's taskresults didn't completed.
+	// We should make sure it the taskresults processing is complete as it will be possible to reference it in the next step.
+	if needReconcilTaskResult {
+		return fmt.Errorf("need reconcil workflowtaskresults")
+	}
```

Comment:
if this isn't an error, we should return it as something else, not an error

Reply:
OK，I will change it.